src/infi/clickhouse_orm/system_models.py

```diff
@@ -0,0 +1,142 @@
+"""
+from .database import Database  # Can't import it globally, due to circular import
+from .fields import *
+from .models import Model
+from .engines import MergeTree
+
+
+class SystemPart(Model):
+    """
+    Contains information about parts of a table in the MergeTree family.
+    This model operates only fields, described in the reference. Other fields are ignored.
+    https://clickhouse.yandex/reference_en.html#system.parts
+    """
+    OPERATIONS = frozenset({'DETACH', 'DROP', 'ATTACH', 'FREEZE', 'FETCH'})
+
+    readonly = True
+
+    database = StringField()  # Name of the database where the table that this part belongs to is located.
+    table = StringField()  # Name of the table that this part belongs to.
+    engine = StringField()  # Name of the table engine, without parameters.
+    partition = StringField()  # Name of the partition, in the format YYYYMM.
+    name = StringField()  # Name of the part.
+    replicated = UInt8Field()  # Whether the part belongs to replicated data.
+
+    # Whether the part is used in a table, or is no longer needed and will be deleted soon.
+    # Inactive parts remain after merging.
+    active = UInt8Field()
+
+    # Number of marks - multiply by the index granularity (usually 8192)
+    # to get the approximate number of rows in the part.
+    marks = UInt64Field()
+
+    bytes = UInt64Field()  # Number of bytes when compressed.
+
+    # Time the directory with the part was modified. Usually corresponds to the part's creation time.
+    modification_time = DateTimeField()
+    remove_time = DateTimeField()  # For inactive parts only - the time when the part became inactive.
+
+    # The number of places where the part is used. A value greater than 2 indicates
+    # that this part participates in queries or merges.
+    refcount = UInt32Field()
+
+    @classmethod
+    def table_name(cls):
+        return 'system.parts'
+
+    """
+    Next methods return SQL for some operations, which can be done with partitions
+    https://clickhouse.yandex/reference_en.html#Manipulations with partitions and parts
+    """
+    def _partition_operation_sql(self, db, operation, settings=None, from_part=None):
+        """
+        Performs some operation over partition
+        :param db: Database object to execute operation on
+        :param operation: Operation to execute from SystemPart.OPERATIONS set
+        :param settings: Settings for executing request to ClickHouse over db.raw() method
+        :return: Operation execution result
+        """
+        operation = operation.upper()
+        assert operation in self.OPERATIONS, "operation must be in [%s]" % ', '.join(self.OPERATIONS)
+        sql = "ALTER TABLE `%s`.`%s` %s PARTITION '%s'" % (db.db_name, self.table, operation, self.partition)
+        if from_part is not None:
+            sql += " FROM %s" % from_part
+        db.raw(sql, settings=settings, stream=False)
+
+    def detach(self, database, settings=None):

```

Comment:
I guess I was not clear enough.

SystemPart.get() and SystemPart.get_active() do need the Database instance, that's fine. And they call database.select(), so when the database creates the SystemPart instances they will get a value for the _database member - just like any other model instance that is loaded from the database! To clarify - database.select will set _database, it will not be a feature specific to the SystemPart class.

So once you load a SystemPart, it will have a _database and there will not be a need to pass the database to operations that work on that instance (freeze, drop, etc.).

I think that this mechanism (the _database member) could open up additional features that work on instances. Consider for example lazy loading - I will be able to do something like:
`Database.select("SELECT id from MyLazyModel", model_class=MyLazyModel) `
which will load instances that only know their ids, and all the rest of the fields are not loaded. And then I can add an instance method MyLazyModel.populate() which will go to the _database and load all the rest of the fields into the model instance.